cmd/registry/patch/artifact.go

```diff
@@ -159,3 +205,66 @@ func applyArtifactPatch(ctx context.Context, client connection.Client, content *
 	}
 	return err
 }
+
+// kindForMimeType returns the message name to be used as the "kind" of the artifact.
+func kindForMimeType(mimeType string) string {
+	parts := strings.Split(mimeType, ".")
+	return parts[len(parts)-1]
+}
+
+// protoMessageForMimeType returns an instance of the message that represents the specified type.
+func protoMessageForMimeType(mimeType string) (proto.Message, error) {
+	messageType := strings.TrimPrefix(mimeType, "application/octet-stream;type=")
+	for k, v := range artifactMessageTypes {
+		if k == messageType {
+			return v(), nil
+		}
+	}
+	return nil, fmt.Errorf("unsupported message type %s", messageType)
+}
+
+// protoMessageForMimeType returns an instance of the message that represents the specified kind.
+func protoMessageForKind(kind string) (proto.Message, error) {
+	for k, v := range artifactMessageTypes {
+		if strings.HasSuffix(k, "."+kind) {
+			return v(), nil
+		}
+	}
+	return nil, fmt.Errorf("unsupported kind %s", kind)
+}
+
+// messageTypeForKind finds the full message type associated with a kind (typically a message name).
+func messageTypeForKind(kind string) string {
```

Comment:
Consider merging this with `MimeTypeForKind` since that's the sole user of this function. Among other benefits, it becomes obvious that an empty type can be returned.

```
func MimeTypeForKind(kind string) string {
	for k := range artifactMessageTypes {
		if strings.HasSuffix(k, "."+kind) {
			return fmt.Sprintf("application/octet-stream;type=%s", k)
		}
	}
	return "application/octet-stream;type="
}
```

Reply:
Done, but changing the fall-through return type to just "application/octet-stream"